skyportal/handlers/api/source.py

```diff
@@ -733,6 +733,38 @@ def get_sources(
             raise ValueError(
                 "Invalid commentsFilter value -- must provide at least one string value"
             )
+
+        comment_query = Comment.select(session.user_or_token).where(
+            Comment.text.in_(comments_filter)
+        )
```

Comment:
Why not filter on all the comment related stuff in one subquery?

Reply:
Should be fixed!